src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -65,10 +66,15 @@ internal sealed override bool OnTryRead(ref Utf8JsonReader reader, Type typeToCo
                         byte[]? propertyNameArray = properties[i].Item4;
                         string? dataExtKey = properties[i].Item5;
 
-                        tempReader = new Utf8JsonReader(
-                            originalSpan.Slice(checked((int)resumptionByteIndex)),
-                            isFinalBlock: true,
-                            state: properties[i].Item2);
+                        tempReader = originalSequence.IsEmpty
```

Comment:
The changes surfaced an unelated bug here: the converter was not taking `Utf8JsonReader` instances reading from `ReadOnlySequence` into account.

Reply:
Is there a test that covers this now?

Reply:
Yes, existing tests started failing because we are no longer copying ReadOnlySequence inputs into a contiguous ReadOnlySpan.